pyspeckit/spectrum/units.py

```diff
@@ -697,9 +704,12 @@ def as_unit(self, unit, frame=None, quiet=True, center_frequency=None,
         elif xtype_check == 'fix':
             self._fix_type()
 
-        if unit in (None,'none'):
```

Comment:
The equivalencies will be either passed in by the user directly (as `equivalencies=...`) or will be set from `center_frequency` (with optional `center_frequency_unit` - remove the `s`) and `velocity_convention`, e.g. as:

```
conventions = {'optical': u.doppler_optical, 'radio': u.doppler_radio, 'relativistic': u.doppler_relativistic}
if not hasattr(center_frequency, 'unit'):
    center_frequency = center_frequency * u.Unit(center_frequency_unit)
self.equivalencies = conventions[velocity_convention](center_frequency)
```
though there will be more test cases needed for when any of these are left as `None`